Build failure: tectonic

### Steps To Reproduce

Steps to reproduce the behavior:
1. `nix shell nixpkgs#tectonic` on aarch64-darwin.

### Build log

```
error: linking with `/nix/store/3fpl3hs6ii820yxp0rbafx5phnmj8ng7-clang-wrapper-16.0.6/bin/cc` failed: exit status: 1
  |
  = note: LC_ALL="C" PATH="/nix/store/2jdssgjk1i4gjz4w9k8nz3831wq7sxdw-rustc-1.73.0/lib/rustlib/aarch64-apple-darwin/bin:/nix/store/46zkgqp0wk7rwd2n7paa561ch5xjkkyz-cargo-1.73.0/bin:/nix/store/51dr5rx4c63z1h9qqlwpfnv73xg8kij6-cargo-auditable-0.6.1/bin:/nix/store/hglj9mby09jrbgxma6q0h0457kkb0a43-pkg-config-wrapper-0.29.2/bin:/nix/store/nghmh9gq10i38c696k1114iq1gc>
  = note: Undefined symbols for architecture arm64:
...
```

https://gist.github.com/AxelTLarsson/59c6d05fd38a65f2d22506e39da1b757

### Additional context

Tectonic fails to build on macOS Sonoma, aarch64-darwin.

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@lluchs @doronbehar 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.1.0, macOS 14.1.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.18.1`
 - channels(axel): `""`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

### Priorities

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/tools/typesetting/tectonic/default.nix
+++ b/pkgs/tools/typesetting/tectonic/default.nix
@@ -30,6 +30,9 @@ rustPlatform.buildRustPackage rec {
   buildInputs = [ icu fontconfig harfbuzz openssl ]
     ++ lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [ ApplicationServices Cocoa Foundation ]);
 
+  # workaround for https://github.com/NixOS/nixpkgs/issues/166205
+  NIX_LDFLAGS = lib.optionalString (stdenv.cc.isClang && stdenv.cc.libcxx != null) " -l${stdenv.cc.libcxx.cxxabi.libName}";
+
   postInstall = lib.optionalString stdenv.isLinux ''
     substituteInPlace dist/appimage/tectonic.desktop \
       --replace Exec=tectonic Exec=$out/bin/tectonic

